File: 06-Trees/InvertBinaryTree.py
class Node:
    def __init__(self, value=0):
        self.value = value
        self.left = None
        self.right= None
'''
    Input
        1
     2    3
   4  5     6  
 7   8

    Output
        1
     3    2
   6    5   4  
         8   7 
'''


def invertTree(root: Node):
    if not root: return None
    tmp = root.left
    root.left = root.right
    root.right = tmp

    invertTree(root.left)
    invertTree(root.right)

    return root

File: 06-Trees/InvertBinaryTree.java
public class InvertBinaryTree {
    public static void main(String[] args) {
        Node root = new Node(1);
        root.left = new Node(3);
        root.right = new Node(4);
        root.left.left = new Node(5);
        root.left.right = new Node(9);
        root.right.left = new Node(6);

        System.out.println(invertTree(root));
    }

    public static Node invertTree(Node root) {
        if(root == null) return null;

        Node tmp = root.left;
        root.left = invertTree(root.right);
        root.right = invertTree(tmp);
        if(root.left != null && root.right != null) {
            System.out.println(root.left.value);
            System.out.println(root.right.value);
        }
        
        return root;
    }
}
